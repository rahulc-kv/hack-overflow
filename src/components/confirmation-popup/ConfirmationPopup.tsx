import React, { FC } from 'react';

import { CloseIcon } from '@assets/icons';
import Button from '@components/button/Button';
import PopUp from '../modal/SideModal';
import COLORS from '@constants/colors';

type ModalProps = {
  buttonOneText: string;
  buttonTwoText: string;
  visibility: boolean;
  message: string;
  onButtonOneClick: () => void;
  onButtonTwoClick: () => void;
  onCloseButtonClick: () => void;
  title: string;
  disabled?: boolean;
  testId?: string;
};

const ConfirmationPopup: FC<ModalProps> = ({
  title = '',
  message = '',
  buttonOneText = '',
  buttonTwoText = '',
  visibility = false,
  onButtonOneClick,
  onButtonTwoClick,
  onCloseButtonClick,
  disabled = false,
  testId = 'confirmationPopup'
}) => {
  const onCloseIconClick = () => {
    onCloseButtonClick();
  };

  return (
    <PopUp isOpen={visibility} handleClose={onCloseButtonClick} testId={testId}>
      <div
        className={`flex justify-center items-center self-center
                    w-full h-full bg-transparent
                     ${
                       visibility
                         ? 'animate-zoom-to-front'
                         : 'animate-zoom-to-back'
                     }`}
        onClick={e => e.stopPropagation()}>
        <div className="flex relative flex-col p-3 max-w-md bg-white rounded-xl">
          <div className="flex absolute top-3 right-3 justify-center items-center w-7 h-7 bg-whiteSmoke rounded-full">
            <Button
              onClick={onCloseIconClick}
              iconStroke={COLORS.DAVY_GREY}
              Icon={CloseIcon}
            />
          </div>
          <p className="px-10 mt-8 text-base font-medium text-center text-blackGreen">
            {title}
          </p>
          <p className="px-10 mt-2 text-sm text-center text-davyGrey">
            {message}
          </p>
          <div className="flex justify-center mt-6 w-full bg-whiteSmoke rounded-xl">
            <Button
              className="my-4 mx-1.5 font-bold text-white bg-secondaryColor rounded-md"
              onClick={onButtonOneClick}
              label={buttonOneText}
              labelClass="py-1.5 px-4 text-sm"
            />
            <Button
              className="my-4 mx-1.5 font-bold text-white bg-primaryColor rounded-md"
              onClick={onButtonTwoClick}
              label={buttonTwoText}
              labelClass="py-1.5 px-4 text-sm"
              disabled={disabled}
            />
          </div>
        </div>
      </div>
    </PopUp>
  );
};

export default ConfirmationPopup;
