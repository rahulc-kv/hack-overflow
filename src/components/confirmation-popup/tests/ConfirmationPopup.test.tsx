import React from 'react';
import { renderIntoDocument } from 'react-dom/test-utils';

import { translate } from '@utils/locale';
import ConfirmationPopup from '../ConfirmationPopup';

test('Confirmation PopUp Rendered', () => {
  const component = renderIntoDocument(
    <ConfirmationPopup
      title={translate('popup.areYouSure')}
      message={translate('popup.connectExchangeWarning')}
      buttonOneText={translate('cancel')}
      buttonTwoText={translate('sourcePage.importTransactions')}
      onButtonOneClick={jest.fn()}
      onButtonTwoClick={jest.fn()}
      onCloseButtonClick={jest.fn()}
      visibility={true}
    />
  );
  expect(component).toMatchSnapshot();
});
