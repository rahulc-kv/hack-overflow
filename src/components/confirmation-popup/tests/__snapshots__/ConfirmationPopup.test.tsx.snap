// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirmation PopUp Rendered 1`] = `null`;
